### DISCUSSION TITLE
Dynamic route for index.js

### DISCUSSION THREAD

**Discussion by marine1079** [vercel/next.js#10714] [Help]
I want my landing page (pages/index.js) to have 0 or 1 param. (kind of user id)
It is for support personal landing page for those have param (user id) for index.js
(For those not have param, I'll just show general form.)

How can I make dynamic route for index.js without making pages/[id].js ?
If I make pages/[id].js, I should manage 2 files (pages/index.js, pages/[id].js) which has almost same code.

(I tried next/router but I could get param by /?id=123 which is not semantic url at all.)

Thanks.

**Comment by jamesmosier** [comment]
This issue might be of assistance, it sounds similar to your question https://github.com/zeit/next.js/issues/10488
